erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -37,11 +37,20 @@ def validate(self):
 	def on_submit(self):
 		self.check_stock_frozen_date()
 		self.actual_amt_check()
+		self.calculate_batch_qty()
 
 		if not self.get("via_landed_cost_voucher"):
 			from erpnext.stock.doctype.serial_no.serial_no import process_serial_no
 			process_serial_no(self)
 
+	def on_cancel(self):
```

Comment:
Actually SLE are not cancelled and `on_cancel` is not called. Instead we update "is_cancelled=Yes" while cancelling a document. So, while calculating batch_qty pass a filter "is_cancelled=No"

Reply:
Ah, I see. So how would I capture any SLE or stock reversals, and recalculate the batch quantity?